doc/controller_configuration/controller_configuration_tutorial.rst

```diff
@@ -30,146 +79,409 @@ The ``controllers.yaml`` configuration file is located in the ``robot_moveit_con
+
+  * Implemented inline in the same header file
+
+As you can see, writing a `controller handle <https://github.com/ros-planning/moveit/blob/master/moveit_core/controller_manager/include/moveit/controller_manager/controller_manager.h#L104>`__ comes down to implementing:
+
+* ``sendTrajectory`` method that translates `moveit_msgs::RobotTrajectory <http://docs.ros.org/en/noetic/api/moveit_msgs/html/msg/RobotTrajectory.html>`_ to a format the controller can understand
+* ``cancelExecution`` method to tell the controller to stop any active trajectories
+* ``waitForExecution`` method that will block the calling thread until the controller finishes or the ``timeout`` is reached
+* ``getLastExecutionStatus`` method that returns the status of the last requested trajectory.
+
+One example *controller handle allocator* plugin implementation is included with MoveIt:
+
+* `joint_trajectory_controller_plugin.cpp <https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_ros_control_interface/src/joint_trajectory_controller_plugin.cpp>`_
+
+The only job of a controller handle allocator is to create a new instance of the controller handle. The following example implements an allocator for a custom controller handle of type ``example::controller_handle_example``:
+
+.. code-block:: c++
+
+  // declares example::controller_handle_example class
+  #include "controller_handle_example.h"
+  #include <moveit_ros_control_interface/ControllerHandle.h>
+  #include <pluginlib/class_list_macros.h>
+
+  namespace example
+  {
+  class controller_handle_allocator_example : public moveit_ros_control_interface::ControllerHandleAllocator
+  {
+  public:
+    moveit_controller_manager::MoveItControllerHandlePtr alloc(const std::string& name,
+                                                               const std::vector<std::string>& resources) override
+    {
+      return std::make_shared<controller_handle_example>(name, std::string("follow_joint_trajectory"));
+    }
+  };
+  }  // namespace example
+
+  PLUGINLIB_EXPORT_CLASS(example::controller_handle_allocator_example,
+                         moveit_ros_control_interface::ControllerHandleAllocator);
+
+
+This example controller handle allocator can be exported by creating a plugin definition file which is then referenced in the ``exports`` section of ``package.xml``:
+
+.. code-block:: XML
+
+    <library path="lib/libtrajectory_controller_example">
+        <class
+            name="example/trajectory_controller_example"
+            type="example::controller_handle_allocator_example"
+            base_class_type="moveit_ros_control_interface::ControllerHandleAllocator"
+        >
+            <description>
+                Example Controller Handle Allocator for MoveIt!
+            </description>
+        </class>
+    </library>
+
+.. note::
+  Replace ``lib/libtrajectory_controller_example`` with your library name by following the same format (prepending ``lib/lib`` to your library name)
```

Comment:
The prefix `lib/` is not needed. 
You explain rather generic plugin usage here. Can't you simply refer to existing pluginlib docs?

Reply:
I checked pluginlib doc at http://wiki.ros.org/pluginlib
There is no mention of whether this prefix is needed or not but it appears in the code example at the end of the page:
```
<library path="lib/librectangle">
  <class name="rviz/Rectangle" type="rectangle_namespace::Rectangle" base_class_type="polygon_namespace::Polygon">
  <description>
  This is a rectangle plugin
  </description>
  </class>
</library>
```

That is why I included this blurb here. Pluginlib docs do not explain why this prefix is needed.
When I removed the prefix, my plugin was no longer being loaded. I don't understand why, and there is no documentation that explains this.